auth_mellon_util.c

```diff
@@ -1540,7 +1612,11 @@ bool am_validate_paos_header(request_rec *r, const char *header)
         goto cleanup;
     }
 
-    /* Split the header on the semicolon character */
+    /*
+     * Split the header on the semicolon character.
+     * There should be 2 items, the first is the version delclaration
+     * and the second is the service and it's options.
+     */
     semicolon_tokens = g_strsplit(header, ";", 0);
```

Comment:
It might be worth checking if Apache has some utility routines for parsing this kind of "maybe-quoted" header strings that can be used by modules.

If one ends up writing a full parser from scratch, I would guess it isn't worth it at the moment. It would be nice if one somehow could ensure that the entire thing doesn't blow up when data that isn't handled by the parser shows up though.